### PR TITLE
Add conftest.py hook for vSphere (#2779)

### PR DESCRIPTION
**Pull request by wusui**
The test_scale_pvc_expand and test_ceph_pod_respin_in_scaled_cluster
tests should not be run on vms.

Signed-off-by: wusui <wusui@redhat.com>

### CODE REVIEW

File: tests/e2e/scale/conftest.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+
+    Args:
+        items: list of collected tests
+
+    """
+    skip_list = [
+        'test_scale_pvc_expand',
+        'test_ceph_pod_respin_in_scaled_cluster'
+    ]
+    if config.ENV_DATA['platform'].lower() == "vsphere":
```
Comment (petr-balogh) [inline tests/e2e/scale/conftest.py:19]: you can use VSPHERE_PLATFORM  from https://github.com/red-hat-storage/ocs-ci/blob/master/ocs_ci/ocs/constants.py#L686

File: tests/e2e/scale/conftest.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
+import logging
+from ocs_ci.framework import config
+
+log = logging.getLogger(__name__)
+
+
+def pytest_collection_modifyitems(items):
+    """
+    A pytest hook to skip certain tests when running on vms.
```
Comment (petr-balogh) [inline tests/e2e/scale/conftest.py:9]: what the vms is stand for here?   Virtual Machines?   If so it should be VMs or whole Virtual Machines.

File: tests/e2e/scale/conftest.py [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        'test_scale_pvc_expand',
+        'test_ceph_pod_respin_in_scaled_cluster'
+    ]
+    if config.ENV_DATA['platform'].lower() == "vsphere":
+        for item in items.copy():
+            for testname in skip_list:
+                if testname in str(item.fspath):
+                    log.info(
+                        f"Test {item} is removed from the collected items"
+                        f" since it does not run on vms"
```
Comment (petr-balogh) [inline tests/e2e/scale/conftest.py:25]: Same question with vms here.

File: tests/e2e/scale/conftest.py (hunk 1 above)
Comment (wusui) [reply tests/e2e/scale/conftest.py]: Changed to use constants.VSPHERE_PLATFORM

File: tests/e2e/scale/conftest.py (hunk 2 above)
Comment (wusui) [reply tests/e2e/scale/conftest.py]: changed to "virtual machine"

File: tests/e2e/scale/conftest.py (hunk 3 above)
Comment (wusui) [reply tests/e2e/scale/conftest.py]: changed to virtual machine

**Comment by ramkiperiy** [comment]
@wusui have you tested this ?

**Comment by wusui** [comment]
I tested this before I checked in the PR.  I will run the PR now too.

**Comment by wusui** [comment]
This is failing with -- Trying to acquire lock on [Label: vSphere-e2e,
Quantity: 1]


It appears that there have been some changes related to this lately,
so I am going to rebase onto the latest and reinstall.


I will also see how this affects the local run-ci issue that I have been seeing.

**Review by ramkiperiy — DISMISSED** [review]

**Review by Avilir — DISMISSED** [review]

**Review by vasukulkarni — APPROVED** [review]

**Review by tiffanyn108 — DISMISSED** [review]

**Comment by tiffanyn108** [comment]
Warren, can you add "test_scale_osds_fill_75%_reboot_workers" to the list since this testcase also running adding capacity.

**Comment by ramkiperiy** [comment]
@wusui How are you triggering this PR validation meaning would like to know the param's you are using. Looks like last PR validation run doesn't skip the tests. I suspect it might be due to the platform param (i.e. default_config.yaml) which is default to AWS, I don't know where to get this input from logs. But the expected tests not skipped.

**Review by ramkiperiy — APPROVED** [review]

**Comment by wusui** [comment]
See Ram's comment.  The change works as expected but the failure is caused by other scale test failures.  That second failure was an invalid run on my part because I thought I could get this to pass by running one of the skipped files but the change does not affect things when an individual run is sepcified.

**Review by vasukulkarni — APPROVED** [review]

**Pull request merged**